ring/ring_test.go

```diff
@@ -2212,6 +2212,14 @@ func TestShuffleShardWithCaching(t *testing.T) {
 	ring.CleanupShuffleShardCache("user")
 	newSubring = ring.ShuffleShard("user", 1)
 	require.False(t, subring == newSubring)
+
+	// If we ask for ALL instances, we get original ring.
+	newSubring = ring.ShuffleShard("user", numLifecyclers)
+	require.True(t, ring == newSubring)
```

Comment:
Can this be `require.Equals(t, ring, newSubring)` to provide more detail errors if the comparison fails?

Reply:
The point is to verify that we do indeed get exact same pointer, and not just another ring that may be equal.

Reply:
> In that case, `require.Same()`

Done.